### DISCUSSION TITLE
How can we reuse Listeners in ver. 14?

### DISCUSSION THREAD

**Discussion by taiwanleaftea** [orchidsoftware/platform#2633] [Q&A]
I have simple Country->State listener, which is used on several screens.
In previous version I've sent table prefix via constructor

`new CountryListener('customer')`

But now listener works via 

`public function handle(Repository $repository, Request $request)` and constructor doesn't get the parameter. 

I've solved this problem adding Input field with class('visually-hidden'), where I can store the prefix. But it looks ugly, because this field occupies a place in the form.

Maybe there is another solution?

**Comment by macit-emre** [comment]
There is no asyncBlablaMethod needed in involved Screen in new setup, so handle method gets the empty Repository class and the request object has the parameters that defined in $targets array. So in your Listener's layout() function, using $this->query->get('blabla') method now is populated with 
1- IN Normal HTTP Request : populated with the returns of the query method 
2- IN Ajax call : the returns of the handle method as a Repository object.
